core/editable.js

```diff
@@ -1618,6 +1635,54 @@
 					return false;
 			};
 		}
+
+		function checkForMultipleList( range ) {
+			// We know that we are in the list so now we must check if there is another one.
+			var listsTypes = [ 'ul', 'ol', 'dl' ],
+				listItems = [ 'dd', 'dt', 'li' ],
+				walker = new CKEDITOR.dom.walker( range ),
+				element = range.collapsed ? range.startContainer : walker.next(),
+				firstRangeEl = range.startContainer,
+				isInclude = false,
+				// Check if the first element from range is an list item.
+				isListItem = CKEDITOR.tools.array.some( listItems, function( listItem ) {
+					var item = firstRangeEl.$.nodeName.toLowerCase();
+					return item === listItem;
+				} );
+
+			if ( !isListItem || firstRangeEl.getPrevious() !== null ) {
+				return;
+			}
+
+			// Walk through all the items in the range to find lists.
+			while ( element ) {
+				var tagName = element.$.nodeName.toLowerCase();
+
+				CKEDITOR.tools.array.forEach( listsTypes, function( listType ) {
+					if ( tagName === listType ) {
+						isInclude = true;
+					}
+				} );
+
+				element = walker.next();
+			}
+
+			if ( isInclude ) {
+				var listParent = null;
+				// Get the parent of the list.
+				CKEDITOR.tools.array.map( listsTypes, function( listType ) {
```

Comment:
I'm not sure why the `map` is used here? Probably some oversight?  Also, this method favors the end elements from `listsTypes`. Are you sure that there will be no errors in more advanced nested samples? Maybe you could provide an additional test case for different nested lists?